includes/class-wc-admin-notes.php

```diff
@@ -58,4 +58,19 @@ public static function get_note( $note_id ) {
 		}
 		return false;
 	}
+
+	/**
+	 * Deletes admin notes with a given name.
+	 *
+	 * @param string $name Name to search for.
+	 */
+	public static function delete_notes_with_name( $name ) {
+		$data_store = WC_Data_Store::load( 'admin-note' );
+		$note_ids   = $data_store->get_notes_with_name( $name );
+		foreach ( (array) $note_ids as $note_id ) {
+			$note = new WC_Admin_Note( $note_id );
+			$note->set_id( $note_id );
```

Comment:
Yeah, if I don't do this, the note isn't fully initialized since we are initializing by ID. That caught me by surprise yesterday as well. Now that I think about it - I will see why proton wrote the data store constructors I reference this way - maybe they have a bug (and the constructor it seems should be fully initializing the note with just an ID passed in.)